### PR TITLE
PQUERY-19 Integrate pquery3 with old pquery framework for PXC

### PR DESCRIPTION
**Pull request by mohitj1988**

### CODE REVIEW

File: pquery-run.sh [hunk 1]
```diff
@@ -475,7 +470,7 @@ pxc_startup(){
     if [[ "$ENCRYPTION_RUN" != 1 ]];then
       sed -i "2i wsrep_provider_options=\"gmcast.listen_addr=tcp://$LADDR1;$WSREP_PROVIDER_OPT\"" ${DATADIR}/n${i}.cnf
 	else
-      sed -i "2i wsrep_provider_options=\"gmcast.listen_addr=tcp://$LADDR1;$WSREP_PROVIDER_OPT;socket.ssl_key=${WORKDIR}/cert/server-key.pem;socket.ssl_cert=${WORKDIR}/cert/server-cert.pem;socket.ssl_ca=${WORKDIR}/cert/ca.pem\"" ${DATADIR}/n${i}.cnf
+      sed -i "2i wsrep_provider_options=\"gmcast.listen_addr=tcp://$LADDR1;$WSREP_PROVIDER_OPT;socket.ssl_key=${WORKDIR}/cert/server-key.pem;socket.ssl_cert=${WORKDIR}/cert/server-cert.pem;socket.ssl_ca=${WORKDIR}/cert/ca.pem;gcache.recover=no\"" ${DATADIR}/n${i}.cnf
```
Comment (rahulmalik87) [inline pquery-run.sh]: remove it if it is not related to this PR.

File: pquery-run.sh [hunk 2]
```diff
@@ -375,11 +375,6 @@ if [[ $PXC -eq 1 ]];then
     echo "master_verify_checksum=on" >> ${BASEDIR}/my.cnf
     echo "binlog_checksum=CRC32" >> ${BASEDIR}/my.cnf
     echo "binlog_encryption=ON" >> ${BASEDIR}/my.cnf
-    echo "innodb_temp_tablespace_encrypt=ON" >> ${BASEDIR}/my.cnf
-    echo "encrypt_tmp_files=ON" >> ${BASEDIR}/my.cnf
-    echo "default_table_encryption=ON" >> ${BASEDIR}/my.cnf
-    echo "innodb_redo_log_encrypt=ON" >> ${BASEDIR}/my.cnf
-    echo "innodb_undo_log_encrypt=ON" >> ${BASEDIR}/my.cnf
     echo "innodb_sys_tablespace_encrypt=ON" >> ${BASEDIR}/my.cnf
```
[the remark this answers is not in the document]
Comment (mohitj1988) [reply pquery-run.sh]: These are normal innodb encryption variables. Should not be hard-coded. Instead we must use pquery to set them ON and OFF while the server load is running

File: pquery-run.sh [hunk 3]
```diff
@@ -1388,24 +1394,27 @@ pquery_test(){
           else
             echoit "Loading metadata from ${WORKDIR}/step_$((${TRIAL}-1)).dll ..."
           fi
+          if [[ ${PXC} -eq 0 && ${GRP_RPL} -eq 0 ]]; then
+            echo "${PQUERY_BIN} ${SQL_FILE} --database=test --threads=${THREADS} --queries-per-thread=${QUERIES_PER_THREAD} --logdir=${RUNDIR}/${TRIAL} --log-failed-queries --user=root --socket=${RUNDIR}/${TRIAL}/socket.sock --seed ${SEED} --step ${TRIAL} --metadata-path ${WORKDIR}/ --seconds ${PQUERY_RUN_TIMEOUT} ${DYNAMIC_QUERY_PARAMETER}";
```
[the remark this answers is not in the document]
Comment (mohitj1988) [reply pquery-run.sh]: Done

File: pquery-run.sh [hunk 4]
```diff
@@ -1622,7 +1631,7 @@ pquery_test(){
       echoit "Pri engine pquery run details:$(grep -i 'SUMMARY.*queries failed' ${RUNDIR}/${TRIAL}/*.sql ${RUNDIR}/${TRIAL}/*.log | sed 's|.*:||')"
       echoit "Sec engine pquery run details:$(grep -i 'SUMMARY.*queries failed' ${RUNDIR}/${TRIAL}/*.sql ${RUNDIR}/${TRIAL}/*.log | sed 's|.*:||')"
     else
-      echoit "pquery run details:$(grep -i 'SUMMARY.*queries failed' ${RUNDIR}/${TRIAL}/*.sql ${RUNDIR}/${TRIAL}/*.log | sed 's|.*:||')"
+      echoit "pquery run details:$(grep -i 'SUMMARY.*queries failed' ${RUNDIR}/${TRIAL}/node1/*.sql ${RUNDIR}/${TRIAL}/node1/*.log | sed 's|.*:||')"
```
[the remark this answers is not in the document]
Comment (mohitj1988) [reply pquery-run.sh]: Multi node support with pquery3 added.

File: pquery-run.sh (hunk 1 above)
Comment (mohitj1988) [reply pquery-run.sh]: Done

File: pquery-run.sh [hunk 5]
```diff
@@ -1329,15 +1339,15 @@ pquery_test(){
...
+            ${PQUERY_BIN} --infile=${INFILE} --database=test --threads=${THREADS} --queries-per-thread=${QUERIES_PER_THREAD} --logdir=${RUNDIR}/${TRIAL} --log-all-queries --log-failed-queries --log-query-duration --user=root --socket=${SOCKET} >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
             PQPID="$!"
           else
             if [[ ${PXC_CLUSTER_RUN} -eq 1 ]];then
               cat ${PXC_CLUSTER_CONFIG} \
                 | sed -e "s|\/tmp|${RUNDIR}\/${TRIAL}|" \
                 | sed -e "s|\/home\/ramesh\/percona-qa|${SCRIPT_PWD}|" \
                 > ${RUNDIR}/${TRIAL}/pquery-cluster.cfg
-              ${PQUERY_BIN} --config-file=${RUNDIR}/${TRIAL}/pquery-cluster.cfg >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
+              ${PQUERY_BIN} --infile=${INFILE} --config-file=${RUNDIR}/${TRIAL}/pquery-cluster.cfg >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
```
Comment (rahulmalik87) [inline pquery-run.sh]: why have you removed inflie?

File: pquery-run.sh [hunk 6]
```diff
@@ -1388,24 +1399,36 @@ pquery_test(){
           else
```
Comment (rahulmalik87) [inline pquery-run.sh]: could you print the seed number?

File: pquery-run.sh (hunk 5 above)
Comment (rahulmalik87) [inline pquery-run.sh]: what is the reason for adding inflie.

File: pquery-run.sh [hunk 7]
```diff
@@ -1388,24 +1399,36 @@ pquery_test(){
...
             echoit "Loading metadata from ${WORKDIR}/step_$((${TRIAL}-1)).dll ..."
           fi
+          if [[ ${PXC} -eq 0 && ${GRP_RPL} -eq 0 ]]; then
+            CMD="${PQUERY_BIN} ${SQL_FILE} --database=test --threads=${THREADS} --queries-per-thread=${QUERIES_PER_THREAD} --logdir=${RUNDIR}/${TRIAL} --log-all-queries --log-failed-queries --user=root --socket=${SOCKET} --seed ${SEED} --step ${TRIAL} --metadata-path ${WORKDIR}/ --seconds ${PQUERY_RUN_TIMEOUT} ${DYNAMIC_QUERY_PARAMETER}"
```
Comment (rahulmalik87) [inline pquery-run.sh]: can you remove SQL_FILE if not consumed by pquery

Comment (rahulmalik87) [reply pquery-run.sh]: --log-all-queries, --log-failed-queried make it configurable.

File: pquery-run.sh (hunk 6 above)
Comment (mohitj1988) [reply pquery-run.sh]: Added a line:
echoit "$CMD" . This will print the seed number

File: pquery-run.sh (hunk 5 above)
Comment (mohitj1988) [reply pquery-run.sh]: It should not be added. Removed.

File: pquery-run.sh (hunk 7 above)
Comment (mohitj1988) [reply pquery-run.sh]: SQL_FILE option is for pquery2, no use for pquery3. Hence removed.
--log-all-queries, --log-failed-queries have been moved to the configuration file and removed from pquery-run.sh

File: pquery-run.sh [hunk 8]
```diff
@@ -1347,20 +1357,21 @@ pquery_test(){
               ${PQUERY_BIN} --config-file=${RUNDIR}/${TRIAL}/pquery-cluster.cfg >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
               PQPID="$!"
             else
-              ${PQUERY_BIN} --infile=${INFILE} --database=test --threads=${THREADS} --queries-per-thread=${QUERIES_PER_THREAD} --logdir=${RUNDIR}/${TRIAL} --log-all-queries --log-failed-queries --log-query-duration --user=root --socket=${RUNDIR}/${TRIAL}/node1/node1_socket.sock >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
+              ${PQUERY_BIN} --infile=${INFILE} --database=test --threads=${THREADS} --queries-per-thread=${QUERIES_PER_THREAD} --logdir=${RUNDIR}/${TRIAL} --log-all-queries --log-failed-queries --user=root --socket=${SOCKET1} >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
```
Comment (rahulmalik87) [inline pquery-run.sh]: can you please add log-query-duration again if it doesn't belong to pquery 3.

File: pquery-run.sh [hunk 9]
```diff
@@ -1388,24 +1399,37 @@ pquery_test(){
           else
             echoit "Loading metadata from ${WORKDIR}/step_$((${TRIAL}-1)).dll ..."
           fi
+          if [[ ${PXC} -eq 0 && ${GRP_RPL} -eq 0 ]]; then
+            CMD="${PQUERY_BIN} --database=test --threads=${THREADS} --queries-per-thread=${QUERIES_PER_THREAD} --logdir=${RUNDIR}/${TRIAL} --user=root --socket=${SOCKET} --seed ${SEED} --step ${TRIAL} --metadata-path ${WORKDIR}/ --seconds ${PQUERY_RUN_TIMEOUT} ${DYNAMIC_QUERY_PARAMETER}"
+            echoit "$CMD"
+            $CMD >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
+            PQPID="$!"
```
Comment (rahulmalik87) [inline pquery-run.sh]: echoit "$CMD"
             $CMD >${RUNDIR}/${TRIAL}/pquery.log 2>&1 &
             PQPID="$!"

are common and can be moved after else

File: pquery-run.sh (hunk 8 above)
Comment (mohitj1988) [reply pquery-run.sh]: Done

File: pquery-run.sh (hunk 9 above)
Comment (mohitj1988) [reply pquery-run.sh]: Done

**Pull request merged**